### DISCUSSION TITLE
Add edges to skeleton post inference

### DISCUSSION THREAD

**Discussion by ab-madabhushi** [talmolab/sleap#1146] [Help!] [answered]
Hello! I am tying to use SLEAP to track jumping spiders. Initially I had 13 nodes in my skeleton, which SLEAP was having a hard time to track. So, I trained SLEAP on a subset of nodes/edges and got it to work. Now, I would want to increase the complexity of my skeleton to include other nodes/body parts gradually. Is there a way to do this in the same project or do I have to track different body parts in different projects? If that is the case is there a way to align co-ordinates of points tracked from different projects into one co-ordinate space?

Thank you for your help!

**Comment by roomrys** [comment, marked as answer]
Hi @ab-madabhushi,

You can make changes to the skeleton in even after you have started labeling. No need for a completely new project (although I would encourage saving checkpoints of your project with _Save As..._). 

The new nodes will only appear as "not visible" and will only appear on the user-labeled instances. For each user-labeled frame you have already labeled with the simple skeleton, you will need to [right click the nodes to toggle the visibility](https://sleap.ai/tutorials/initial-labeling.html?highlight=labeling) of the new node.

Let us know if you run into any trouble.

Thanks,
Liezl

**Reply by ab-madabhushi** [reply]
Thank you very much for your reply!
Yes, sorry I realized I should have been more clear. So, I constructed a subset skeleton (with only face and forelegs), trained a model to track this, ran the inference and am satisfied with how SLEAP tracked these points. Now, I added the new nodes i.e., thorax and pedipalps and labelled these (in the frames that had user-labeled instances), trained the model and asked it to predict instances in a subset of frames. Now, the problem I am having is I am not able to detect the frames on which predictions were carried. I would want to look at the frames with predicted instances, correct the predictions for SLEAP to be able to track more complicated skeleton (i.e., skeleton with face-forelegs-thorax-pedipalps) and run a new inference to track all these points (instead of just the simple skeleton of face and forelegs). 

Would this be possible i.e., once SLEAP runs the inference to track a subset of points, is it possible to add new points to the skeleton, get SLEAP to predict instances of this more complicated skeleton, and run a new inference to track the more complicated skeleton?

I am not sure if explained my problem clearly. Sorry for being unclear but, let me know if this made sense and if that would be possible. If yes, could you please direct me to how to do this?

Thanks again!

**Reply by roomrys** [reply]
Hi @ab-madabhushi,

Yes, it is possible to add new points to the skeleton, get SLEAP to predict instances of this more complicated skeleton, and run a new inference to track the more complicated skeleton.

1. You need to make sure that all user-labeled frames have the new nodes labeled (and marked as visible where the body part is visible). 
2. You then need to re-run training to generate a model that know to look for another body part(s).
3. Run inference

Since you are having trouble finding the new predictions, I would suggest that you [Generate Labeling suggestions and predict on these suggestions](https://sleap.ai/tutorials/initial-labeling.html?highlight=labeling%20suggestion#selecting-frames-to-label). You can also remove all predictions (Labels > Delete All Predictions) so that you know any predictions will be from the modified labeled with the complex skeleton.

Once you have the model trained that knows to look for the extra body parts, you can re-use this model (without re-training) to run inference and generate predictions that SHOULD include the extra nodes.

Thanks,
Liezl

**Reply by ab-madabhushi** [reply]
That makes sense. I will try this!

Thank you!
Abhinav
